src/nodes/accessors/VelocityNode.js

```diff
@@ -0,0 +1,104 @@
+import { addNodeClass } from '../core/Node.js';
+import TempNode from '../core/TempNode.js';
+import { modelWorldMatrix } from './ModelNode.js';
+import { positionLocal } from './PositionNode.js';
+import { nodeImmutable } from '../shadernode/ShaderNode.js';
+import { NodeUpdateType } from '../core/constants.js';
+import { Matrix4 } from '../../math/Matrix4.js';
+import { uniform } from '../core/UniformNode.js';
+import { sub } from '../math/OperatorNode.js';
+
+const _worldMatrixCache = new WeakMap();
+
+class VelocityNode extends TempNode {
+
+	constructor() {
+
+		super( 'vec2' );
+
+		this.updateType = NodeUpdateType.OBJECT;
+		this.updateAfterType = NodeUpdateType.OBJECT;
+
+		this.previousProjectionViewMatrix = uniform( new Matrix4() );
+		this.currentProjectionViewMatrix = uniform( new Matrix4() );
+		this.previousModelMatrix = uniform( new Matrix4() );
+
+		this.firstFrame = true;
+
+	}
+
+	update( frame ) {
+
+		const camera = frame.camera;
+		const object = frame.object;
+
+		if ( this.firstFrame ) {
+
+			this.firstFrame = false;
+
+			this.currentProjectionViewMatrix.value.multiplyMatrices( camera.projectionMatrix, camera.matrixWorldInverse );
+			this.previousProjectionViewMatrix.value.copy( this.currentProjectionViewMatrix.value );
+
+			this.previousModelMatrix.value.copy( object.matrixWorld );
+
+		} else {
+
+			this.previousProjectionViewMatrix.value.copy( this.currentProjectionViewMatrix.value );
+			this.currentProjectionViewMatrix.value.multiplyMatrices( camera.projectionMatrix, camera.matrixWorldInverse );
+
+			const previousModelMatrix = getPreviousModelMatrix( object );
+			this.previousModelMatrix.value.copy( previousModelMatrix );
+
+		}
+
+	}
+
+	updateAfter( frame ) {
+
+		const object = frame.object;
+
+		const previousModelMatrix = getPreviousModelMatrix( object );
+
+		previousModelMatrix.copy( object.matrixWorld );
+
+	}
+
+	setup( /*builder*/ ) {
+
+		this.firstFrame = true;
+
+		const clipPositionCurrent = this.currentProjectionViewMatrix.mul( modelWorldMatrix ).mul( positionLocal ).toVar();
+		const clipPositionPrevious = this.previousProjectionViewMatrix.mul( this.previousModelMatrix ).mul( positionLocal ).toVar();
+
+		const ndcPositionCurrent = clipPositionCurrent.xy.div( clipPositionCurrent.w );
+		const ndcPositionPrevious = clipPositionPrevious.xy.div( clipPositionPrevious.w );
+
+		let velocity = sub( ndcPositionCurrent, ndcPositionPrevious ).mul( 0.5 );
+		velocity = velocity.mul( 0.5 ).add( 0.5 );
```

Comment:
The value range in the texture is `[0,1]` where `(0.5,0.5)` means no velocity. If we want to sample the velocity and use it in a computation, we can restore the NDC velocity like so:
```js
const sceneVelocity = scenePass.getTextureNode( 'velocity' );
const sample = sceneVelocity.uv( uvNode ).rg; 
const velocity = sample.mul( 2.0 ).sub( 1.0 ); // range should now [-1, 1], NDC
```
@WestLangley  Does this all look plausible to you?

Reply:
Wait, I think the target range should be `[-2, 2]`. 

So:
```js
let velocity = sample.mul( 2.0 ).sub( 1.0 );
velocity = velocity.mul( 2.0 );
```

Reply:
I think you need to pack `velocity.xy` to an RGBA color. (Or just pack the x-component to RG for the purposes of this example.) See the channels example.

Reply:
A comparable packing like in the original isn't required since we are using half float textures right now.

Reply:
The TSL in `VelocityNode` should be correct, imo.

My question was about the restore of the velocity and a think `[-2, 2]` should be the target range. The velocity is the result of a vector subtraction and if you subtract extreme points, you will have result vectors in the range `[-2, 2]`.

Reply:
Got it... But something does not seem right to me. Not sure yet what I am missing...

Reply:
I'll try to implement a per-object motion blur the next days. So if something does not work, I'll hopefully realize it soon. But given that all other resources I have checked so far perform basically the same computation, I believe we are on the right track.

Reply:
>you will have result vectors in the value range [-2, 2].

Yes. And that is the reason for the `mul( 0.5 )` -- to remap to [-1, 1].

It seems the units of velocity are pixels per frame. If you want to restore that, I think you can scale it by whatever scaling factor makes sense for your app.